### PR TITLE
Create traps.yml

### PR DESCRIPTION
**Pull request by Mesverrum**
@i3149 I wanted to confirm a few things to make sure what I'm doing in here is actually workable in ktranslate.

I've got the OID and names of a few traps from a MIB document I was working from, but I haven't seen any of the varbind oids yet so I was imagining that if we leave drop undefined as false I would at least get the name of the trap and later on we can fill in the events stanza, see lines 7-18 without the events compared to 19-31.  

The other piece I wanted to ask about was that in the packet capture I saw several of the oids were including index numbers of tabular entries, is there some different syntax we need to use to deal with that?  For example the varbind oid would be .1.3.6.1.4.1.12196.13.1.1.1, but a set of 3 incoming traps would come in as .1.3.6.1.4.1.12196.13.1.1.1.8, .1.3.6.1.4.1.12196.13.1.1.1.10, .1.3.6.1.4.1.12196.13.1.1.1.32.

### CODE REVIEW

File: profiles/kentik_snmp/cisco/traps.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    drop_undefined: false
+  - trap_oid: 1.3.6.1.6.3.1.1.5.3
+    trap_name: linkDown
+    drop_undefined: false
+  - trap_oid: 1.3.6.1.6.3.1.1.5.4
+    trap_name: linkUp
+    drop_undefined: false
+  - trap_oid: 1.3.6.1.6.3.1.1.5.5
+    trap_name: authenticationFailure
+    drop_undefined: false
```
Comment (thezackm) [inline profiles/kentik_snmp/cisco/traps.yml:21]: curious: why would we not put these into a `/_general/traps.yml` profile?

**Comment by i3149** [comment]
> @i3149 I wanted to confirm a few things to make sure what I'm doing in here is actually workable in ktranslate.
> 
> I've got the OID and names of a few traps from a MIB document I was working from, but I haven't seen any of the varbind oids yet so I was imagining that if we leave drop undefined as false I would at least get the name of the trap and later on we can fill in the events stanza, see lines 7-18 without the events compared to 19-31.

Correct: 

```
				toid := v.Value.(string)
				trap = s.mibdb.GetTrap(toid)
				dst.CustomStr["TrapOID"] = toid
				if trap != nil {
					dst.CustomStr["TrapName"] = trap.Name
				}
```

So `TrapOID` is always set, with the human readable `TrapName` optional. 

> 
> The other piece I wanted to ask about was that in the packet capture I saw several of the oids were including index numbers of tabular entries, is there some different syntax we need to use to deal with that? For example the varbind oid would be .1.3.6.1.4.1.12196.13.1.1.1, but a set of 3 incoming traps would come in as .1.3.6.1.4.1.12196.13.1.1.1.8, .1.3.6.1.4.1.12196.13.1.1.1.10, .1.3.6.1.4.1.12196.13.1.1.1.32.

That's a good question. Its not something that we've had to deal with before and probably isn't supported now. Can you send me the pcap file to test on?

File: profiles/kentik_snmp/cisco/traps.yml (hunk 1 above)
Comment (Mesverrum) [reply profiles/kentik_snmp/cisco/traps.yml:21]: Looking at the customer data where I got this from the trap OID is generic, but its also included varbinds that are specific to Cisco mibs.  I had the same thought until I saw that Cisco specific stuff.  I think its similar in concept to the way the cisco entity sensor mib gets all mingled with the generic snmp entity mibs.

**Comment by Mesverrum** [comment]
Sent you the data I have in Slack, but I think the solution is we will need to come up with a syntax for tabular traps where we say everything up to 1.3.6.4...x is the trapoid to match against, and then whatever value is after that is the Index that just needs to be decorated into the payload.  Hopefully that makes sense and isn't too hard.  Not sure if we are doing an exact match right now but it almost feels like just allowing syntax like this and adding in the logic to extract the index would give us what we needed
`trap_oid: 1.3.6.1.6.3.1.1.5.5.*`

**Review by thezackm — APPROVED** [review]

**Pull request merged**